infobserve/sources/github.py

```diff
@@ -0,0 +1,119 @@
+            event_list (list) : A list of GistEvent Objects.
+        """
+
+        headers: Dict = {
+            "User-Agent": 'Infobserver',
+            "Accept": self.API_VERSION,
+            "Authorization": f'token {self._oauth_token}'
+        }
+
+        async with aiohttp.ClientSession(headers=headers) as session:
+            resp = await session.get(self._uri)
+            APP_LOGGER.debug("GithubSource Response Headers: %s", resp.headers)
+            github_events = await resp.json()
+            event_list: List[GithubEvent] = []
+            tasks = []
+
+            APP_LOGGER.debug("GithubSource: %s Fetched Recent %s Public Events", self.name, len(github_events))
+
+            if self._index_cache:
+                cached_ids = await self._index_cache.query_index_cache()
+                # At the moment only PushEvent support we should productize more logic into helper classes
+                # To support all the event types.
+                github_events = [x for x in github_events if x["id"] not in cached_ids and x["type"] == "PushEvent"]
+            else:
+                github_events = [x for x in github_events if x["type"] == "PushEvent"]
+
+            APP_LOGGER.debug("Github Push Events number: %s", len(github_events))
+
+            for event in github_events:
+                # Create GistEvent objects and create io intensive tasks.
+                ge = GithubEvent(event, session)
+
+                try:
+                    event_list.append(ge)
+                    tasks.append(asyncio.create_task(ge.get_raw_content()))
+                except asyncio.TimeoutError:
+                    APP_LOGGER.warning("Dropped event with id:%s url not valid", ge.id)
+
+            # Update the index_cache.
+            if self._index_cache:
+                await self._index_cache.update_index_cache([x["id"] for x in github_events])
+
+            # Fetch the commits async.
+            await asyncio.gather(*tasks)
+
+            # Create an event for each commit
+            for event in event_list:
+                await event.get_raw_content()
+
+            commit_event_list: List[GithubEvent] = [x for x in event.commit_raw_content() for event in event_list]
+
+            APP_LOGGER.debug("%s Github Commits send for processing", len(commit_event_list))
+            return commit_event_list
+
+    async def fetch_events_scheduled(self, queue: ProcessingQueue):
+        """
+        Call the fetch_events method on a schedule.
+
+        Arguments:
+           queue (ProcessingQueue): A processing queue to enqueue the events.
+        """
+        while True:
+            try:
+                events: List[GithubEvent] = await self.fetch_events()
+                for event in events:
+                    await queue.queue_event(event)
+            except aiohttp.client_exceptions.ClientPayloadError:
+                APP_LOGGER.warning("There was an error retrieving the payload will retry in next cycle.")
```

Comment:
Does `ClientPayloadError` not contain any useful information? Maybe we could log its message as well?